vcd/datasource_vcd_nsxt_ip_set.go

```diff
@@ -57,17 +63,57 @@ func datasourceVcdNsxtIpSet() *schema.Resource {
 func datasourceVcdNsxtIpSetRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	vcdClient := meta.(*VCDClient)
 
-	nsxtEdgeGateway, err := vcdClient.GetNsxtEdgeGatewayFromResourceById(d, "edge_gateway_id")
+	org, err := vcdClient.GetOrgFromResource(d)
 	if err != nil {
-		return diag.Errorf(errorUnableToFindEdgeGateway, err)
+		return diag.Errorf("[nsxt ip set read] error retrieving Org: %s", err)
 	}
 
-	// Name uniqueness is enforced by VCD for types.FirewallGroupTypeIpSet
-	ipSet, err := nsxtEdgeGateway.GetNsxtFirewallGroupByName(d.Get("name").(string), types.FirewallGroupTypeIpSet)
-	if err != nil {
-		return diag.Errorf("error getting NSX-T IP Set with Name '%s': %s", d.Get("name").(string), err)
+	ipSetName := d.Get("name").(string)
+	edgeGatewayId := d.Get("edge_gateway_id").(string)
+
+	var ipSet *govcd.NsxtFirewallGroup
+	var parentVdcOrVdcGroupId string
+
+	if ipSetName != "" && edgeGatewayId != "" {
```

Comment:
Just a nit suggestion, to get rid of 1 level of identation:
```suggestion
	if ipSetName == "" || edgeGatewayId == "" {
	    return diag.Errorf("error - not all parameters specified for NSX-T IP Set lookup")
	}
	
	// Lookup Edge Gateway to know parent VDC or VDC Group
	anyEdgeGateway, err := org.GetAnyTypeEdgeGatewayById(edgeGatewayId)
	...
```

Reply:
nice, changed.